language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/CodeGenRunner.scala

```diff
@@ -3,23 +3,108 @@
 
 package com.daml.lf.codegen
 
-import java.nio.file.{Files, Path, StandardOpenOption}
+import java.nio.file.{Files, Path}
 import java.util.concurrent.atomic.AtomicInteger
-import java.util.concurrent.{Executors, ThreadFactory, TimeUnit}
+import java.util.concurrent.{Executors, ThreadFactory}
+
 import com.daml.lf.archive.DarParser
-import com.daml.lf.codegen.backend.Backend
-import com.daml.lf.codegen.backend.java.JavaBackend
+import com.daml.lf.codegen.backend.java.inner.{ClassForType, DecoderClass, fullyQualifiedName}
 import com.daml.lf.codegen.conf.{Conf, PackageReference}
-import com.daml.lf.data.ImmArray
 import com.daml.lf.data.Ref.PackageId
-import com.daml.lf.iface.reader.{Errors, InterfaceReader}
-import com.daml.lf.iface.{Type => _, _}
+import com.daml.lf.iface.reader.{InterfaceReader, Errors}
+import com.daml.lf.iface.{EnvironmentInterface, InterfaceType, Interface}
+import com.squareup.javapoet.{JavaFile, ClassName}
 import com.typesafe.scalalogging.StrictLogging
-import org.slf4j.{Logger, LoggerFactory}
+import org.slf4j.{LoggerFactory, Logger, MDC}
 
 import scala.collection.immutable.Map
-import scala.concurrent.duration.Duration
-import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContextExecutorService, Await, ExecutionContext, Future}
+
+private final class CodeGenRunner(
```

Comment:
All of this used to be part of `Backend`, which was supposed to be a language-agnostic interface for the codegen.

1. This never came to be
2. The interface itself got polluted with Java-specific concepts

No reason to have that around anymore.